src/sentry/models/authenticator.py

```diff
@@ -56,33 +56,44 @@ def all_interfaces_for_user(self, user, return_missing=False):
         interfaces are returned even if not enabled.
         """
         _sort = lambda x: sorted(x, key=lambda x: (x.type == 0, x.type))
-        rv = [x.interface for x in Authenticator.objects.filter(user=user)
-              if x.interface.is_available]
-        if not return_missing:
-            return _sort(rv)
 
-        rvm = dict(AUTHENTICATOR_INTERFACES)
-        for iface_cls in rv:
-            rvm.pop(iface_cls.interface_id, None)
-        for iface_cls in rvm.itervalues():
-            iface = iface_cls()
-            if iface.is_available:
-                rv.append(iface)
+        # Collect interfaces user is enrolled in
+        ifaces = [x.interface for x in Authenticator.objects.filter(
+            user=user,
+            type__in=[a.type for a in available_authenticators()],
```

Comment:
> Probably currently that's the case since you can't remove that one and it's always available.

It returns no results anyways, so it's fine even if we had no results.